test/teiserver/game/balancer_server_test.exs

```diff
@@ -35,8 +40,176 @@ defmodule Teiserver.Game.BalancerServerTest do
     result = BalancerServer.calculate_team_size(team_count, players)
     assert result == 4
   end
+end
 
-  defp create_fake_players(count) do
-    1..count |> Enum.map(fn _ -> %{} end)
+defmodule Teiserver.Game.BalancerServerTest do
+  @moduledoc false
+  use Teiserver.DataCase, async: false
+  import Teiserver.Support.Polling, only: [poll_until_nil: 1]
+
+  @moduletag :balance_test
+  alias Teiserver.Game.BalancerServer
+
+  def user_fixture(), do: make_user(%{"permissions" => []})
+
+  def make_user(params \\ %{}) do
+    requested_user = %{
+      "name" => params["name"] || "Test",
+      "email" => params["email"] || "email@email#{:rand.uniform(999_999_999_999)}",
+      "colour" => params["colour"] || "#00AA00",
+      "icon" => params["icon"] || "fa-solid fa-user",
+      "password" => params["password"] || "password",
+      "password_confirmation" => params["password"] || "password",
+      "data" => params["data"] || %{}
+    }
+
+    {:ok, _u} =
+      Teiserver.Account.create_user(requested_user)
+  end
+
+  @spec make_users_with_ranks_and_parties(
+          [non_neg_integer()],
+          [non_neg_integer()]
+        ) ::
+          list()
+
+  def make_users_with_ranks_and_parties(list_of_ranks, list_of_parties) do
+    users =
+      Enum.with_index(list_of_ranks)
+      |> Enum.map(fn {_user, index} ->
+        %{
+          "name" => "user" <> to_string(index + 1) <> "_" <> to_string(ExULID.ULID.generate()),
+          "email" => to_string(ExULID.ULID.generate()) <> "@example.com",
+          "permissions" => []
+        }
+      end)
+      |> Enum.map(fn x -> make_user(x) end)
+      # unwrap user creation response
+      |> Enum.map(fn {:ok, reply} -> reply end)
+      |> Enum.map(fn x -> Map.put(x, :userid, x.id) end)
```

Comment:
I don't think you care about the username at all, so you shouldn't need the index to create the usernames and emails, the random ones from the fixture are enough.

Also, you don't need to chain multiple maps like that, you can have one single function, and also combine it with the loop just below:
```elixir
Enum.map(list_of_ranks, fn rank ->
  {:ok, user} = make_user()
  update_user_stat(user.id, %{:rank => rank})
end)
```

Also, why do you need to set `:userid` on the created user?